cmd/nerdctl/container_run_mount_windows_test.go

```diff
@@ -0,0 +1,126 @@
+	t.Parallel()
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	rwDir, err := os.MkdirTemp(t.TempDir(), "rw")
+	if err != nil {
+		t.Fatal(err)
+	}
+	roDir, err := os.MkdirTemp(t.TempDir(), "ro")
+	if err != nil {
+		t.Fatal(err)
+	}
+	rwVolName := tID + "-rw"
+	roVolName := tID + "-ro"
+	for _, v := range []string{rwVolName, roVolName} {
+		defer base.Cmd("volume", "rm", "-f", v).Run()
+		base.Cmd("volume", "create", v).AssertOK()
+	}
+
+	containerName := tID
+	defer base.Cmd("rm", "-f", containerName).AssertOK()
+	base.Cmd("run",
+		"-d",
+		"--name", containerName,
+		"-v", fmt.Sprintf("%s:C:/mnt1", rwDir),
+		"-v", fmt.Sprintf("%s:C:/mnt2:ro", roDir),
+		"-v", fmt.Sprintf("%s:C:/mnt3", rwVolName),
+		"-v", fmt.Sprintf("%s:C:/mnt4:ro", roVolName),
+		testutil.CommonImage,
+		"ping localhost -t",
+	).AssertOK()
+
+	base.Cmd("exec", containerName, "cmd", "/c", "echo -n str1 > C:/mnt1/file1").AssertOK()
+	base.Cmd("exec", containerName, "cmd", "/c", "echo -n str1 > C:/mnt2/file1").AssertFail()
+	base.Cmd("exec", containerName, "cmd", "/c", "echo -n str1 > C:/mnt3/file1").AssertOK()
+	base.Cmd("exec", containerName, "cmd", "/c", "echo -n str1 > C:/mnt4/file1").AssertFail()
+	base.Cmd("rm", "-f", containerName).AssertOK()
+
+	base.Cmd("run",
+		"--rm",
+		"-v", fmt.Sprintf("%s:C:/mnt1", rwDir),
+		"-v", fmt.Sprintf("%s:C:/mnt3", rwVolName),
+		testutil.CommonImage,
+		"cat", "C:/mnt1/file1", "C:/mnt3/file3",
+	).AssertOutExactly("str1str3")
+	base.Cmd("run",
+		"--rm",
+		"-v", fmt.Sprintf("%s:C:/mnt3/mnt1", rwDir),
+		"-v", fmt.Sprintf("%s:C:/mnt3", rwVolName),
+		testutil.CommonImage,
+		"cat", "C:/mnt3/mnt1/file1", "C:/mnt3/file3",
+	).AssertOutExactly("str1str3")
+}
+
+func TestRunVolumeRelativePath(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+
+	containerName := tID
+	defer base.Cmd("rm", "-f", containerName).AssertOK()
+
+	base.Cmd("run",
+		"-d",
+		"--name", containerName,
+		"-v", "./mnt:C:/mnt1",
+		testutil.CommonImage,
+		"top",
+	).AssertOK()
+
+	// Destination cannot be a relative epath
+	base.Cmd("run",
+		"-d",
+		"--name", containerName,
+		"-v", "./mnt:./mnt1",
+		testutil.CommonImage,
+		"top",
+	).AssertFail()
+}
+
+func TestRunAnonymousVolume(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	base.Cmd("run", "--rm", "-v", "C:/foo", testutil.CommonImage,
+		"mountpoint", "-q", "C:/foo").AssertOK()
+	base.Cmd("run", "--rm", "-v", "TestVolume", testutil.CommonImage).AssertOK()
+	base.Cmd("run", "--rm", "-v", "TestVolume2:C:/TestVolumes", testutil.CommonImage).AssertOK()
+}
+
+func TestRunNamedPipeVolume(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	base.Cmd("run", "--rm", "-v", `\\.\pipe\containerd-containerd`, testutil.CommonImage).AssertOK()
+}
+
+func TestInvalidPath(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	base.Cmd("run", "--rm", "-v", `InvalidPathC:\TestVolume:C:\Mount`, testutil.CommonImage).AssertFail()
```

Comment:
Please consider adding more potentially invalid cases such as:

* no colons in source/target/neither
* multiple colons being present
* anything other formatting issue which may cross your mind